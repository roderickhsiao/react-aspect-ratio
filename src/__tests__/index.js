import React from 'react';
import renderer from 'react-test-renderer';

import AspectRatio from '../index';

describe('Aspect Ratio', () => {
  it('should render wrapper for children', () => {
    const innerImage = <img src="https://foo.bar" alt="demo" />;
    const node = renderer.create(<AspectRatio ratio="4/3">{innerImage}</AspectRatio>).toJSON();
    expect(node).toMatchSnapshot();
  });

  it('should support number as props', () => {
    const innerImage = <img src="https://foo.bar" alt="demo" />;
    const node = renderer.create(<AspectRatio ratio={0.75}>{innerImage}</AspectRatio>).toJSON();
    expect(node).toMatchSnapshot();
  });

  it('custom style willl be adpoted', () => {
    const innerImage = <img src="https://foo.bar" alt="demo" />;
    const node = renderer
      .create(
        <AspectRatio ratio="4/3" style={{ color: '#fff' }}>
          {innerImage}
        </AspectRatio>
      )
      .toJSON();

    expect(node).toMatchSnapshot();
  });
});
