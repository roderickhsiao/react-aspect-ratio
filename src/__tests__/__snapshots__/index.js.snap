// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aspect Ratio custom style willl be adpoted 1`] = `
<div
  className="react-aspect-ratio-placeholder"
  style={
    Object {
      "--aspect-ratio": "(4/3)",
      "color": "#fff",
    }
  }
>
  <img
    alt="demo"
    src="https://foo.bar"
  />
</div>
`;

exports[`Aspect Ratio should render wrapper for children 1`] = `
<div
  className="react-aspect-ratio-placeholder"
  style={
    Object {
      "--aspect-ratio": "(4/3)",
    }
  }
>
  <img
    alt="demo"
    src="https://foo.bar"
  />
</div>
`;

exports[`Aspect Ratio should support number as props 1`] = `
<div
  className="react-aspect-ratio-placeholder"
  style={
    Object {
      "--aspect-ratio": "(0.75)",
    }
  }
>
  <img
    alt="demo"
    src="https://foo.bar"
  />
</div>
`;
